test/check-application

```diff
@@ -2513,6 +2513,75 @@ class TestFiles(testlib.MachineCase):
         b.click(".pf-v5-c-modal-box__footer button.pf-m-secondary")
         b.wait_not_present(".file-editor-modal")
 
+    def testCreateFile(self) -> None:
+        b = self.browser
+        m = self.machine
+
+        self.enter_files()
+
+        def open_create_file_modal() -> None:
+            b.mouse("#files-card-parent", "contextmenu")
+            b.click(".contextMenu button:contains('Create file')")
+            b.wait_visible(".file-create-modal")
+            b.wait_text(".pf-v5-c-modal-box__title-text", "Create file")
+            b.wait_visible("button.pf-m-primary:disabled")
+
+        # Error cases
+        # cancel does not create a file
+        open_create_file_modal()
+        b.set_input_text("#file-name", "test.txt")
+        b.set_input_text(".file-create-modal textarea", "content")
+        b.wait_visible("button.pf-m-primary:not(:disabled)")
+        b.click(".pf-v5-c-modal-box__footer button.pf-m-link")  # cancel
+        b.wait_not_present(".pf-v5-c-modal-box")
+        m.execute("! test -f test.txt")
+
+        # file already exists
+        m.execute("runuser -u admin touch /home/admin/test.txt")
+        b.wait_visible("[data-item='test.txt']")
+        open_create_file_modal()
+        b.set_input_text("#file-name", "test.txt")
+        b.wait_text(".file-create-modal .pf-v5-c-helper-text__item-text", "File exists")
+        b.set_input_text("#file-name", "/test.txt")
+        b.wait_text(".file-create-modal .pf-v5-c-helper-text__item-text", "Name cannot include a /.")
+
+        # Create a file as admin, we are superuser and /home owned by admin
+        b.set_input_text("#file-name", "notes.txt")
+        b.set_input_text(".file-create-modal textarea", "content")
+        b.click("button.pf-m-primary")
+
+        b.wait_visible(".file-editor-modal")
```

Comment:
Same question as above: Why does this happen? The creation dialog already sets an initial content, why is the user being thrown into the editor dialog here?

Reply:
As explained on matrix this could use discussion on what we want. Either just create a file and then no editing or create + edit.

The creation and editing is split up as that avoids introducing a lot of complexity in the editor dialog. My old implementation would also show a brief re-render as it used two textarea's one for creating and one for editing.

Reply:
OK, so the idea is "one dialog for creation and its permissions, then directly another edit dialog", check.